codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/serialize/QuerySerializerGenerator.kt

```diff
@@ -171,11 +174,20 @@ abstract class QuerySerializerGenerator(codegenContext: CodegenContext) : Struct
                 "Input" to structureSymbol,
                 *codegenScope,
             ) {
-                serializeStructureInner(context)
+                context.copy(writerExpression = "writer", valueExpression = ValueExpression.Reference("input"))
+                    .also { inner ->
+                        for (member in inner.shape.members()) {
+                            val memberContext = MemberContext.structMember(inner, member, symbolProvider)
+                            structWriter(memberContext) { writerExpression ->
+                                serializeMember(memberContext.copy(writerExpression = writerExpression))
+                            }
+                        }
+                    }
```

Comment:
This looks identical to `serializeStructureInner`?

Reply:
It is. I think I broke it out b/c I wanted to change something but then it didn't end up being needed and I just forgot to switch back to using `serializeStructureInner`. I'll update this.